### PR TITLE
Pd 3363 get guids from solr and data from Redis

### PR DESCRIPTION
**Pull request by vigowei**
## What's this PR do?
Adding logic to get guids from solr first then pull more information from Redis or MySQL if needed. 

## Screenshots (if appropriate)
N/A

## Where should the reviewer start?
seo/search_backend.py line#302

## How should this be manually tested?
 job search on microsites should still work,

## Any background context you want to provide?
To have it pull data from outside of solr, add fields to SEO_REDIS_FIELDS.
Still fixing some tests.

### CODE REVIEW

File: seo/search_backend.py [hunk 1]
```diff
@@ -299,33 +301,41 @@ def search(self, query_string, sort_by=None, start_offset=0, end_offset=None,
...
+            # fields come in as list, set or string
+            if not isinstance(fields, (list, set)):
+                fields = fields.split()
+            requested_fields_in_redis = set(fields) & set(settings.SEO_REDIS_FIELDS)
+            fields = set(fields) - requested_fields_in_redis
+            if requested_fields_in_redis:
+                fields.add('guid')
+            fields = ' '.join(fields)
             kwargs['fl'] = fields
-       # This code was causing sort_by to break, but we're keeping it as a
```
Comment (almathew) [inline seo/search_backend.py]: @JLMcLaugh's old code can be deleted.

File: seo/search_backend.py [hunk 2]
```diff
@@ -462,6 +472,12 @@ def search(self, query_string, sort_by=None, start_offset=0, end_offset=None,
             self.log.error("Failed to query Solr using '%s': %s", query_string, e)
             raw_results = EmptyResults()
 
+        if requested_fields_in_redis:
```
Comment (almathew) [inline seo/search_backend.py]: Is there a way we can have this rely on a flag too? There are some situations (mostly via CLI) in which we're going to bypass Redis regardless of settings.

Comment (vigowei) [inline seo/search_backend.py]: how about we allow something like `multi_src=True` to be passed as a kwarg to search?

Comment (almathew) [inline seo/search_backend.py]: Sure, that's exactly what I was hoping for.

**Comment by almathew** [comment]
Closing while @vigowei implements the flag.

**Comment by vigowei** [comment]
@almathew reopening with some updates including the flag option to allow bypassing Redis.  I added an`SEO_LOG_ES_HOSTS` option to default settings, it won't break anything if no valid ES url(s) are supplied, it'll just fall back to python logger which is probably stderr or stdout.  
I'm thinking that part (seo/utils/log.py) can be improved and generalized more, so all modules, not just seo can leverage it for logging... probably worth a new ticket.

**Comment by almathew** [comment]
Now you have merge conflicts - probably because of some of the changes I made with logging.

**Comment by vigowei** [comment]
@almathew just saw your comment, it was something in the settings file and I took care of it

**Comment by almathew** [comment]
I'm starting a much more in-depth review of this, but @mklauber and @JLMcLaugh your input would also be welcomed.

**Comment by almathew** [comment]
PR is not playing nicely with new solr schema.

**Pull request closed without merging**